VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleProductsController.cs

```diff
@@ -297,6 +303,46 @@ public IHttpActionResult Delete([FromUri] string[] ids)
             return StatusCode(HttpStatusCode.NoContent);
         }
 
+
+        /// <summary>
+        /// Bulk deletes the specified items by ids.
+        /// </summary>
+        /// <param name="productSearchCriteria"></param>
+        [HttpPost]
+        [Route("delete")]
+        [ResponseType(typeof(void))]
+        public IHttpActionResult BulkDelete(ProductSearchCriteria productSearchCriteria)
+        {
+            const int deleteBatchSize = 50;
+
+            var idsToDelete = productSearchCriteria.ObjectIds?.ToList() ?? new List<string>();
+
+            if (idsToDelete.IsNullOrEmpty())
+            {
+                var searchCriteria = productSearchCriteria.ToSearchCriteria();
+                // Any pagination for deleting should be managed at back-end. 
+                searchCriteria.Take = deleteBatchSize;
+                searchCriteria.Skip = 0;
```

Comment:
I think this code should be in the method where deletion happens.

Reply:
Do you mean "GetIdsToDelete" method? I do not want to add "deleteBatchSize" parameter to the method. And I think "GetIdsToDelete" should return ids and shouldn't manage pagination properties.

Reply:
`deleteBatchSize` should be constant on the class level, not in the method.

Reply:
moved